[SDEM-PFEM] Updating DEM-PFEM coupling to new Swimming DEM standard

ipouplana opened the pull request.

jgonzalezusua committed:
Merge branch 'SDEM/developing-new-element-format' into coupling-sdem-pfem

jgonzalezusua committed:
Analysis stage added to PfemFluidDynamicsApplication (on going)

jgonzalezusua committed:
Merge branch 'master' into coupling-sdem-pfem

jgonzalezusua committed:
Merge branch 'SDEM/developing-new-element-format' into coupling-sdem-pfem

jgonzalezusua committed:
Merge branch 'SDEM/developing-new-element-format' into coupling-sdem-pfem

jgonzalezusua committed:
Added rotation process

jgonzalezusua committed:
Merge branch 'SDEM/developing-new-element-format' into coupling-sdem-pfem

jgonzalezusua committed:
Merge branch SDEM/developing-new-element-format into coupling-sdem-pfem

jgonzalezusua committed:
Merge branch 'master' into coupling-sdem-pfem

jgonzalezusua committed:
Removing pfem_fluid_solver_.py and print in DEM_analysis_stage.py

jgonzalezusua committed:
Merge branch 'master' into coupling-sdem-pfem

jgonzalezusua committed:
Removing Lift Force in dictionary and avoid to check Cond Number

jgonzalezusua committed:
Merge branch 'master' into coupling-sdem-pfem

jgonzalezusua committed:
Merge remote-tracking branch 'origin/SDEM/debug-correcting-mistakes-hydro-law' into coupling-sdem-pfem

jgonzalezusua committed:
Remove 'LIFT_FORCE' from variables to print and calculate Nodal Area

jgonzalezusua committed:
Merge branch 'SDEM/debug-correcting-mistakes-hydro-law' into coupling-sdem-pfem

jgonzalezusua committed:
Adding DEM-PFEM "one-way" test

jgonzalezusua committed:
Merge branch 'master' into coupling-sdem-pfem

jgonzalezusua committed:
Removing CalculateNodalAreaProcess from pfem_fluid_dynamics_analysis

jgonzalezusua committed:
Clean up

GuillermoCasas commented on applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py:
```diff
@@ -194,7 +194,7 @@ def SetBetaParameters(self):
         Add("laplacian_calculation_type").SetInt(0)
         Add("faxen_terms_type").SetInt(0)
-        Add("material_acceleration_calculation_type").SetInt(1)
+        Add("material_acceleration_calculation_type").SetInt(8)
```
Modify your own analysis

GuillermoCasas commented on applications/SwimmingDEMApplication/tests/NightTests.py:
```diff
@@ -41,7 +46,8 @@ def SetTestSuite(suites):
           candelier_with_history_test,
-          candelier_with_history_hinsberg_test
+          candelier_with_history_hinsberg_test,
+          Test
```
Use similar nomenclature

GuillermoCasas commented on applications/SwimmingDEMApplication/python_scripts/DEM_coupled_pfem_fluid_dynamics_analysis.py:
```diff
@@ -0,0 +1,64 @@
+
+class DEMCoupledPFEMFluidDynamicsAnalysis(PfemFluidDynamicsAnalysis):
+
+    def __init__(self, model, parameters=None, pp=None):
```
Here you should be passing the whole project_parameters and only pass to the parent the project_parameters["fluid_parameters"] bit. See [https://github.com/KratosMultiphysics/Kratos/blob/SDEM/debugging-and-cleaning-analysis/applications/SwimmingDEMApplication/python_scripts/DEM_coupled_fluid_dynamics_analysis.py](url).

GuillermoCasas reviewed (changes requested):
See comments

jgonzalezusua committed:
Fixing minor errors

philbucher commented on applications/PfemFluidDynamicsApplication/python_scripts/pfem_fluid_dynamics_analysis.py:
```diff
@@ -0,0 +1,393 @@
+import DEM_coupled_pfem_fluid_dynamics_analysis as fluid_solution
+import pfem_fluid_solver_analysis as pfem_solver
+
+from analysis_stage import AnalysisStage
```
it is more future-proof to do the following:
```suggestion
from KratosMultiphysics.analysis_stage import AnalysisStage
```

GuillermoCasas reviewed (dismissed):
I am approving, but please wait for my next pull request to apply more changes to the application

jgonzalezusua committed:
Merge branch 'SDEM/debugging-and-cleaning-analysis' into coupling-sdem-pfem

jgonzalezusua committed:
Fixing minor errors

jgonzalezusua committed:
Merge branch 'master' into coupling-sdem-pfem

GuillermoCasas commented on applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py:
```diff
@@ -5,36 +5,28 @@
-        #self.pp.domain_size = self.fluid_solution.ProjectParameters["problem_data"]["domain_size"].GetInt()
-        #super(Algorithm,self).SetCouplingParameters(varying_parameters)
+        self.project_parameters["material_acceleration_calculation_type"].SetInt(8)
+        #self.vars_man.Dt = self.project_parameters['fluid_parameters']["problem_data"]["time_step"]
```
Please remove comments

GuillermoCasas commented on applications/SwimmingDEMApplication/python_scripts/swimming_DEM_procedures.py:
```diff
@@ -78,6 +78,7 @@ def InitializeVariablesWithNonZeroValues(parameters, fluid_model_part, balls_mod
         SetModelPartSolutionStepValue(fluid_model_part, FLUID_FRACTION_OLD, 1.0)
     if checker.ModelPartHasNodalVariableOrNot(balls_model_part, FLUID_FRACTION_PROJECTED):
         SetModelPartSolutionStepValue(balls_model_part, FLUID_FRACTION_PROJECTED, 1.0)
+        SetModelPartSolutionStepValue(balls_model_part, BASSET_FORCE, Vector([0.]*3))
```
This should not be needed... Why is it not initially 0? Could you investigate this?

GuillermoCasas commented on applications/SwimmingDEMApplication/python_scripts/variables_management.py:
```diff
@@ -199,7 +199,7 @@ def ConstructListsOfVariables(self, parameters):
             self.dem_vars += [VIRTUAL_MASS_FORCE]
 
-        if parameters["basset_force_type"].GetInt() > 0 and parameters["add_each_hydro_force_option"].GetBool():
+        if True or parameters["basset_force_type"].GetInt() > 0 and parameters["add_each_hydro_force_option"].GetBool():
```
For the moment just remove the 'if' and add it always please

GuillermoCasas commented on applications/SwimmingDEMApplication/python_scripts/variables_management.py:
```diff
@@ -332,6 +332,7 @@ def ConstructListsOfResultsToPrint(self, parameters):
         self.variables_to_print_in_file = ["DRAG_FORCE", "LIFT_FORCE", "BUOYANCY", "VELOCITY"]
 
         self.dem_printing_vars = []
+
```
remove space

GuillermoCasas committed:
Removing need to add BASSET_FORCE always and some warnings

Now all tests, including the new PFEM-DEM one run fine. Some warnings from PFEM App remain when running the latter.
